sdk/core/Azure.Core.Amqp/src/AmqpAddress.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Amqp
+{
+    /// <summary>
+    /// Represents the AMQP address.
+    /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-address-string"/>.
+    /// </summary>
+    public struct AmqpAddress : IEquatable<AmqpAddress>
```

Comment:
Should we override the equality and inequality operators because we're overriding equals and implementing `IEquatable<T>`?

Reply:
Not sure if this is needed as I will just be using reference equality for those, but will add.

Reply:
Generally, you want the behavior of `Equals` and the operators to align;  if you're doing structural equality in the `Equals` method, it's confusing for the operator to be doing a referential equality.    As a public type, and as an equatable struct, it would seem that aligning on structural would be the expectation from developers using the type.     

Maybe I'm misunderstanding your comment?

Reply:
That is a good point. I was looking at other structs where Equals used some sort of value equality and == still used Reference.